radicle/src/node.rs

```diff
@@ -40,8 +42,9 @@ pub struct Connection {
 }
 
 impl Connection {
-    pub fn connect<P: AsRef<Path>>(path: P) -> Result<Self, io::Error> {
-        let stream = UnixStream::connect(path)?;
+    pub fn connect<P: AsRef<Path>>(path: P) -> Result<Self, Error> {
+        let stream = UnixStream::connect(path)
+            .map_err(|e| Error::WithContext("failed to connect to node", e))?;
```

Comment:
Is this instead of the `io::Error`? I'd recommend doing:
```rust
pub enum Error {
  #[error("failed to connect to node: {0}")]
  Connect(#[source] io::Error),
}
```

I like this because we have specific errors for the scenarios that can "go wrong". I also wonder about adding the error in the display if it's the source. Aren't there ways to unravel those to get full displays in a pretty way?

Reply:
Yeah, makes sense.